### PR TITLE
Separate production from test code in storage modules

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

File: gradle/projects.main.properties [hunk 1]
```diff
@@ -44,17 +44,23 @@ nessie-tasks-service-impl=tasks/service/impl
 nessie-versioned-spi=versioned/spi
 nessie-versioned-storage-batching=versioned/storage/batching
 nessie-versioned-storage-bigtable=versioned/storage/bigtable
+nessie-versioned-storage-bigtable-tests=versioned/storage/bigtable-tests
```
Comment (snazy) [inline gradle/projects.main.properties:47]: Those new modules would need to be added to the bom as well

**Review by snazy — COMMENTED** [review]
The test classes don't need to be moved - feels a bit awkward to have the tests in another module. It's sufficient to add the new `*-tests` artifacts as a dependency in the main module for test-fixtures or intTest.

**Comment by adutra** [comment]
> The test classes don't need to be moved - feels a bit awkward to have the tests in another module. It's sufficient to add the new `*-tests` artifacts as a dependency in the main module for test-fixtures or intTest.

I recall a similar discussion a while ago, where we concluded that it would be awkward as well to have:

```
main module[intTest] 
  \--> test module[main]
    \--> main module[main]
``` 

Don't you think it's cleaner to have test infra code + tests together in the test module?

**Comment by snazy** [comment]
> > The test classes don't need to be moved - feels a bit awkward to have the tests in another module. It's sufficient to add the new `*-tests` artifacts as a dependency in the main module for test-fixtures or intTest.
> 
> I recall a similar discussion a while ago, where we concluded that it would be awkward as well to have:
> 
> ```
> main module[intTest] 
>   \--> test module[main]
>     \--> main module[main]
> ```
> 
> Don't you think it's cleaner to have test infra code + tests together in the test module?

The thing is that the _test classes_ get distributed to other modules as well, which isn't nice IMO. Supporting code however is okay.

BTW: we do exactly what you quoted (`module_a(intTest) -> other_module -> module_a(main)`) in other places.

File: versioned/storage/jdbc/src/main/resources/META-INF/services/org.projectnessie.versioned.storage.testextension.BackendTestFactory [hunk 2]
```diff
@@ -1,18 +0,0 @@
-#
```
Comment (snazy) [inline versioned/storage/jdbc/src/main/resources/META-INF/services/org.projectnessie.versioned.storage.testextension.BackendTestFactory:1]: Why is this file deleted?

Comment (adutra) [reply versioned/storage/jdbc/src/main/resources/META-INF/services/org.projectnessie.versioned.storage.testextension.BackendTestFactory:1]: In fact it was moved to `versioned/storage/jdbc-tests/src/main/resources/META-INF/services/org.projectnessie.versioned.storage.testextension.BackendTestFactory` – but since the contents changed significantly (because of package renamings), git considers this as a removal + addition.

Comment (snazy) [reply versioned/storage/jdbc/src/main/resources/META-INF/services/org.projectnessie.versioned.storage.testextension.BackendTestFactory:1]: Ah, thanks for the clarification.

**Review by snazy — DISMISSED** [review]
Big PR, but rather boring, so +1

**Comment by snazy** [comment]
BTW: there's a conflicting change
